components/builder-worker/src/runner/publisher.rs

```diff
@@ -48,20 +49,17 @@ impl Publisher {
         let ident = archive.ident().unwrap();
         let target = archive.target().unwrap();
 
-        match retry(RETRIES,
-                    RETRY_WAIT,
-                    || client.x_put_package(archive, auth_token),
-                    |res| {
-                        match *res {
-                            Ok(_) => true,
-                            Err(_) => {
-                                let msg = format!("Upload {}: {:?}", ident, res);
-                                debug!("{}", msg);
-                                logger.log(&msg);
-                                false
-                            }
-                        }
-                    }) {
+        match retry(wait_for(RETRY_WAIT, RETRIES), || {
+                  match client.x_put_package(archive, auth_token) {
+                      Ok(_) => Ok(()),
+                      Err(err) => {
+                          let msg = format!("Upload {}: {:?}", ident, err);
+                          debug!("{}", msg);
+                          logger.log(&msg);
+                          Err(err)
+                      }
+                  }
```

Comment:
Since `client.x_put_package` returns `Result<()>` already, `Ok(_) => Ok(())` is redundant. This could be simplified and made more like the above usage in components/builder-worker/src/runner/mod.rs:
```rust
                  let res = client.x_put_package(archive, auth_token);

                  if let Err(ref err) = res {
                      let msg = format!("Upload {}: {:?}", ident, err);
                      debug!("{}", msg);
                      logger.log(&msg);
                  }

                  res
```
Similarly for `client.create_channel` below.